app.rb

```diff
@@ -1,130 +1,145 @@
     end
   end
 
-  def create_a_book
-    print 'Title: '
-    title = gets.chomp
+  def create_teacher(age, specialization, name, permission)
+    @people << Teacher.new(age, specialization, name, permission)
+    puts 'Teacher created successfully'
+  end
+
+  # create a person
+  def create_person
+    puts 'Do you want to create a student or a Teacher?'
+    choice = get_user_input_as_int('Please Enter 1 for a student or 2 for a Teacher: ')
+    name = get_user_input('Please Enter Name: ')
+    age = get_user_input_as_int('Please Enter Age: ')
+
+    case choice
+    # when person being created student
+    when 1
+      create_student(name, age)
+
+    # when person being created is a teacher
+    when 2
+      specialization = get_user_input('Teacher\'s specialization: ')
+      # create a teacher
+      create_teacher(age, specialization, name, 'true')
+    else
+      puts 'Invalid Input! Should be 1 or 2'
+    end
+  end
 
-    print 'Author: '
-    author = gets.chomp
+  # create a book
+  def create_book
+    title = get_user_input('Please Enter Book Title: ')
+    author = get_user_input('Please Enter Book Author: ')
 
     @books.push(Book.new(title, author))
-    puts 'Book successfully created!'
-    library_menu
+    puts 'Book created successfully'
   end
 
-  def create_a_person
-    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
-    person_no = gets.chomp
-
-    case person_no
-    when '1'
-      create_a_student
-    when '2'
-      create_a_teacher
+  # create a rental
+  def create_rental
+    if @books.empty?
+      puts 'There are no books to rent'
     else
-      puts 'Invalid selection'
-      puts ''
+      puts 'Please select a book number from the following list:'
+      list_of_books
+      book_number = get_user_input_as_int('Book number: ') - 1
+
+      puts 'Please select a person by number (and not person_id) from the following list:'
+      list_of_people
+      person_number = get_user_input_as_int('Person Number: ') - 1
+      entered_date = get_user_input('Please Enter Today\'s Date, use format(YYYY-MM-DD eg.2022-12-13): ')
+
+      # now you have all neccessary variables create and add a rental
+      @rentals << Rental.new(entered_date, @books[book_number], @people[person_number])
+
+      puts 'Rental created successfully'
     end
-    library_menu
   end
 
-  def create_a_rental
-    rent_book
+  def rentals_by_person_id
+    if !@people.empty? && !@rentals.empty?
+      puts 'Please select person id from the following list of people:'
+      list_of_people
+      id = get_user_input_as_int('Person id: ')
+
+      @rentals.each do |rental|
+        if rental.person.id == id
+          puts "Book Name: #{rental.book.title} Book Author: #{rental.book.author} Date: #{rental.date}"
+        else
+          puts "No book found by person id #{id}"
+        end
+      end
+    else
+      puts 'There are no rentals'
```

Comment:
- [ ] @Nejjari001 and @Nowembabazi great work on refactoring however its best practice to ensure all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input.

Reply:
- [ ] Also check on your rental display method got the response below when running the app, showing I have no rentals but still displaying the rentals 
![image](https://user-images.githubusercontent.com/61408860/214271537-47854a53-a460-4675-acb6-0219e3e9d756.png)
![image](https://user-images.githubusercontent.com/61408860/214271687-8b621908-6b7f-4db1-986b-573bb67f227a.png)